### PR TITLE
activate timeline plot

### PR DESCRIPTION
**Pull request by moontrip**
This will activate timeline plot by setting timeline plugin. In this initial PR, it is activated for SAM's X-Y Relationship as shown in the screenshot below. 

sam-timeline00

sam-timeline01

### CODE REVIEW

**Comment by moontrip** [comment]
- It is not clearly decided whether this timeline plot will be hosted at ClinEpiDB's eda (pass app)

- We will need a new SVG icon for timeplot. @asizemore, I think you made other nice SVG icons, so perhaps it will belong to you? 😃

**Comment by asizemore** [comment]
yes, i'll make an issue for creating the icon 👍

**Review by dmfalke — COMMENTED** [review]
This is looking good, but I think we can clean it up a bit more. I would change the `timeseries` plugin declaration to something like this:

```ts
          // activate timeplot Viz
          timeseries: vizWithCustomizedGetRequest(
            // use timelineplotVisualization and set showMarginalHistogram to be true
            vizWithOptions(
              lineplotVisualization
                .withOptions({
                  showMarginalHistogram: true
                })
                .withSelectorIcon(LinePlotSVG)
            ),
            lineplotRequest
          ),
```

This will allow you to remove the `timelineplotVisualization` const from `packages/libs/eda/src/lib/core/components/visualizations/implementations/LineplotVisualization.tsx`.

It will also allow you to remove the addition of `showMarginalHistogram` from the `LayoutOptions` interface. You will need to add the property to `packages/libs/eda/src/lib/core/components/visualizations/implementations/LineplotVisualization.tsx`, line 270.

Let me know if you have questions.

**Comment by moontrip** [comment]
Thanks @dmfalke for your review and suggestion. I will try to address them 👍

**Comment by moontrip** [comment]
@dmfalke I have applied your feedbacks in the new commit except LayoutOptions one. I tried to follow your suggestion but it causes several type issues, so I just leave it as is. Anyway, I will update this PR again when Ann will make a new SVG icon for timeline Viz.

**Comment by dmfalke** [comment]
Thanks, @moontrip!

Here is a diff with the additional changes I had in mind:

```patch
diff --git a/packages/libs/eda/src/lib/core/components/layouts/types.ts b/packages/libs/eda/src/lib/core/components/layouts/types.ts
index 33293dbdd5..826f3ffd4e 100644
--- a/packages/libs/eda/src/lib/core/components/layouts/types.ts
+++ b/packages/libs/eda/src/lib/core/components/layouts/types.ts
@@ -17,8 +17,6 @@ export interface LayoutOptions {
   layoutComponent?: (props: LayoutProps) => JSX.Element;
   hideShowMissingnessToggle?: boolean;
   hideFacetInputs?: boolean;
-  // considering marginal histogram
-  showMarginalHistogram?: boolean;
 }
 
 export interface TitleOptions {
diff --git a/packages/libs/eda/src/lib/core/components/visualizations/implementations/LineplotVisualization.tsx b/packages/libs/eda/src/lib/core/components/visualizations/implementations/LineplotVisualization.tsx
index 83070f6e62..3be1fd6b75 100755
--- a/packages/libs/eda/src/lib/core/components/visualizations/implementations/LineplotVisualization.tsx
+++ b/packages/libs/eda/src/lib/core/components/visualizations/implementations/LineplotVisualization.tsx
@@ -259,7 +259,9 @@ interface Options
       LineplotConfig,
       FloatingLineplotExtraProps,
       LineplotRequestParams
-    > {}
+    > {
+      showMarginalHistogram?: boolean;
+    }
 
 function LineplotViz(props: VisualizationProps<Options>) {
   const {
```

**Comment by moontrip** [comment]
> Thanks, @moontrip!
> 
> Here is a diff with the additional changes I had in mind:
> 
> ```diff
> diff --git a/packages/libs/eda/src/lib/core/components/layouts/types.ts b/packages/libs/eda/src/lib/core/components/layouts/types.ts
> index 33293dbdd5..826f3ffd4e 100644
> --- a/packages/libs/eda/src/lib/core/components/layouts/types.ts
> +++ b/packages/libs/eda/src/lib/core/components/layouts/types.ts
> @@ -17,8 +17,6 @@ export interface LayoutOptions {
>    layoutComponent?: (props: LayoutProps) => JSX.Element;
>    hideShowMissingnessToggle?: boolean;
>    hideFacetInputs?: boolean;
> -  // considering marginal histogram
> -  showMarginalHistogram?: boolean;
>  }
>  
>  export interface TitleOptions {
> diff --git a/packages/libs/eda/src/lib/core/components/visualizations/implementations/LineplotVisualization.tsx b/packages/libs/eda/src/lib/core/components/visualizations/implementations/LineplotVisualization.tsx
> index 83070f6e62..3be1fd6b75 100755
> --- a/packages/libs/eda/src/lib/core/components/visualizations/implementations/LineplotVisualization.tsx
> +++ b/packages/libs/eda/src/lib/core/components/visualizations/implementations/LineplotVisualization.tsx
> @@ -259,7 +259,9 @@ interface Options
>        LineplotConfig,
>        FloatingLineplotExtraProps,
>        LineplotRequestParams
> -    > {}
> +    > {
> +      showMarginalHistogram?: boolean;
> +    }
>  
>  function LineplotViz(props: VisualizationProps<Options>) {
>    const {
> ```

@dmfalke Thank you for your information. Actually, I tried the same thing but somehow it threw several type errors at other components 🤔

File: packages/libs/eda/src/lib/map/analysis/hooks/standaloneVizPlugins.ts [hunk 1]
```diff
@@ -108,6 +106,19 @@ export function useStandaloneVizPlugins({
             vizWithOptions(lineplotVisualization),
             lineplotRequest
           ),
+          // activate timeline Viz
+          timeseries: vizWithCustomizedGetRequest(
+            vizWithOptions(
```
Comment (bobular) [inline packages/libs/eda/src/lib/map/analysis/hooks/standaloneVizPlugins.ts:111]: Don't need double-nested `vizWithOptions` here

**Review by bobular — APPROVED** [review]
I think it's fine to leave `showMarginalHistogram` in the `LayoutOptions` interface. I guess it's specific to the lineplot and could go in `LineplotVisualization` as Dave suggests, but no strong opinion. Other plots could gain a marginal histogram in the future, I suppose.

Just one trivial thing to address.

**Comment by bobular** [comment]
I also noticed quite a few new (to me, not this PR) comments that don't really add information for the reader, here's one example:

```
  // define showMarginalHistogram
  const showMarginalHistogram = options?.showMarginalHistogram ?? false;
```

**Comment by asizemore** [comment]
> Other plots could gain a marginal histogram in the future, I suppose.

Yes indeed! Marginal boxplots have been requested a few times as well. I doubt this will be the only marginal plot we add!

File: packages/libs/eda/src/lib/map/analysis/hooks/standaloneVizPlugins.ts (hunk 1 above)
Comment (moontrip) [reply packages/libs/eda/src/lib/map/analysis/hooks/standaloneVizPlugins.ts:111]: Oops, my mistake!

**Comment by moontrip** [comment]
> I also noticed quite a few new (to me, not this PR) comments that don't really add information for the reader, here's one example:
> 
> ```
>   // define showMarginalHistogram
>   const showMarginalHistogram = options?.showMarginalHistogram ?? false;
> ```

Yes it was commented at the very early stage of the implementation for my record. Removed.

**Comment by dmfalke** [comment]
> @dmfalke Thank you for your information. Actually, I tried the same thing before, but somehow it threw several type errors at other components: that was the reason why I made the previous comment [#322 (comment)](https://github.com/VEuPathDB/web-monorepo/pull/322#issuecomment-1610275806), so I didn't change LayoutOptions thinking

Interesting, it all compiled fine for me. I guess it's okay to leave it as-is, as @bobular suggests.

**Review by dmfalke — APPROVED** [review]

**Comment by moontrip** [comment]
SVG icon will be addressed in the other PR, so merging this PR for now

**Pull request merged**